test_service.py

```diff
@@ -0,0 +1,83 @@
+
+# Copyright 2016, 2017 John J. Rofrano. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Test cases for the Recommendations Service
+Test cases can be run with:
+  nosetests
+  coverage report -m
+"""
+
+import logging
+import unittest
+import json
+from flask_api import status    # HTTP Status Codes
+from models import DataValidationError
+import service
+
+######################################################################
+#  T E S T   C A S E S
+######################################################################
+class TestRecommendationsservice(unittest.TestCase):
+    """ Recommendations service Tests """
+
+    @classmethod
+    def setUpClass(cls):
+        """ Run once before all tests """
+        service.app.debug = False
+        service.initialize_logging(logging.ERROR)
+
+    def setUp(self):
+        """ Runs before each test """
+        service.Recommendation.remove_all()
+        self.app = service.app.test_client()
+
+    def tearDown(self):
+        """ Runs after each test """
+        service.Recommendation.remove_all()
+
+    def test_create_recommendation(self):
+        """Create a recommendation"""
+        # save the current number of recommendations for later comparison
+        recommendation_count = self.get_recommendation_count()
+        # add a new recommendation
+        new_recommendation = {'id': 0, 'product_id': 6, 'recommended_product_id': 7, 'recommendation_type': "up-sell", 'likes': 10}
+        data = json.dumps(new_recommendation)
+        resp = self.app.post('/recommendations', data=data, content_type='application/json')
+        self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+        # Make sure location header is set
+        location = resp.headers.get('Location', None)
+        self.assertTrue(location != None)
+        # Check the data is correct
+        new_json = json.loads(resp.data)
+        self.assertEqual(new_json['product_id'], 6)
+        # check that count has gone up and includes 4
+        resp = self.app.get('/recommendations')
```

Comment:
I think you can replace line 67-70 by just one line like this:
`new_recommendation_count = self.get_recommendation_count()`
reuse the code and make it more clear to read!

Reply:
I tried but nosetest didn't pass because I needed the *data* that I got back for *assertIn*